pkg/services/sqlstore/migrator/migrator.go

```diff
@@ -96,20 +95,28 @@ func (mg *Migrator) GetMigrationLog() (map[string]MigrationLog, error) {
 }
 
 func (mg *Migrator) Start() (err error) {
-	mg.Logger.Info("Locking database")
-	if err := database.CasRestoreOnErr(&mg.isLocked, false, true, ErrDatabaseIsLocked, mg.Dialect.Lock); err != nil {
-		mg.Logger.Error("Failed to lock database", "error", err)
-		return err
-	}
-
-	defer func() {
-		mg.Logger.Info("Unlocking database")
-		err = database.CasRestoreOnErr(&mg.isLocked, true, false, ErrDatabaseIsUnlocked, mg.Dialect.Unlock)
-		if err != nil {
-			mg.Logger.Error("Failed to unlock database", "error", err)
+	return mg.InTransaction(func(sess *xorm.Session) error {
+		mg.Logger.Info("Locking database")
+		if err := casRestoreOnErr(&mg.isLocked, false, true, ErrMigratorIsLocked, mg.Dialect.Lock, sess); err != nil {
+			mg.Logger.Error("Failed to lock database", "error", err)
+			return err
 		}
-	}()
 
+		defer func() {
+			mg.Logger.Info("Unlocking database")
+			unlockErr := casRestoreOnErr(&mg.isLocked, true, false, ErrMigratorIsUnlocked, mg.Dialect.Unlock, sess)
+			if unlockErr != nil {
+				mg.Logger.Error("Failed to unlock database", "error", unlockErr)
+			}
+			sess.Close()
```

Comment:
Just making a sanity check here - do we need to close `sess` manually here, or will `mg.InTransaction` take care of this?

If we need to always close it manually, then we need to do the same on line 101, in the error handling of `casRestoreOnErr` or maybe just put `defer sess.Close()` at the top of the `InTransaction` callback.

Reply:
sorry, my mistake `InTransaction()` takes care of it; I just forgot to remove it when I copied.

Reply:
Okay, thanks for the clarification!